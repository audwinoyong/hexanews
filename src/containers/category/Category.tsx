/**
 * Display All article that contain in certain category screen
 */

import * as React from 'react'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { connect } from 'react-redux'
import { map } from 'lodash'
import { Dispatch } from 'redux'

import ArticleRow from '../../components/articleRow/ArticleRow'
import CategoryHeader from '../../components/categoryHeader/CategoryHeader'
import styles from './styles'
import selector, { StateProps } from './selector'

type Props = StateProps & {
  dispatch: Dispatch<any>
}

export class Category extends React.Component<Props> {
  public render() {
    const { categoryArticles, categoryTitle, dispatch } = this.props
    return (
      <div style={styles.container}>
        <CategoryHeader />
        <Typography
          component='h1'
          gutterBottom
          variant='headline'
        >
          {categoryTitle}
        </Typography>
        <Typography
          component='h2'
          gutterBottom
          variant='subheading'
        >
          Behind the scenes
        </Typography>
        <Grid container spacing={24}>
          <Grid item xs={8} style={styles.latestArticleContainer}>
            {map(categoryArticles, (article) => (
              <ArticleRow
                article={article}
                dispatch={dispatch}
                key={article._id}
              />
            ))}
          </Grid>
        </Grid>
      </div>
    )
  }
}

export default connect(selector)(Category as any)