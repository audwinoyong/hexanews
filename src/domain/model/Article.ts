/**
 * Model for the article.
 */
export type Article = {
  title: string,
  description: string
  category?: string
  authorId?: string
}