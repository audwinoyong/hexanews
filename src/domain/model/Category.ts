/**
 * Model for category object in our application.
 */

export type Category = {
  _id?: string,
  title: string,
  description: string,
}