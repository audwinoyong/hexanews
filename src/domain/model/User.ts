/**
 * Model for the user.
 */
export type User = {
  _id: string,
  email: string,
  name: string,
  description: string,
  createdAt: string,
}