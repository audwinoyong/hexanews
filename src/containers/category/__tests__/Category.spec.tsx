import * as React from 'react'
import * as renderer from 'react-test-renderer'
import * as Enzyme from 'enzyme'
import { BrowserRouter } from 'react-router-dom'
import * as Adapter from 'enzyme-adapter-react-16'

import { Category } from '../Category'
import { Article } from '../../../domain/model/Article'

Enzyme.configure({ adapter: new Adapter() })

const mockArticles: Article[] = [
  {
    _id: '1',
    title: 'abc',
    description: 'test',
    category: 'technology',
    authorId: 'test',
  },
]

const mockUser = {
  _id: '1',
  createdAt: 'created',
  description: 'desc',
  email: 'user@email.com',
  name: 'user'
}

const mockCategory = {
  title: 'tech',
  description: 'test',
}

describe('Category', () => {

  it('renders Category component', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Category
          categoryArticles={mockArticles}
          category={mockCategory}
          categories={[mockCategory]}
          dispatch={jest.fn()}
          users={[mockUser]}
        />
      </BrowserRouter>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})