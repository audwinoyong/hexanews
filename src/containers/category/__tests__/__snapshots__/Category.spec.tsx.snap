// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category renders Category component 1`] = `
<div
  style={
    Object {
      "padding": "5%",
      "paddingLeft": "5%",
      "paddingTop": "0%",
    }
  }
>
  <div
    style={
      Object {
        "flexGrow": 1,
        "overflow": "auto",
      }
    }
  >
    <header
      className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-7"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "boxShadow": "none",
          "paddingRight": 18,
        }
      }
    >
      <div
        className="MuiToolbar-root-36 MuiToolbar-regular-38 MuiToolbar-gutters-37"
      >
        <span
          className="MuiTypography-root-40 MuiTypography-button-51 MuiTypography-colorPrimary-60"
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <a
            href="/category/technology"
            onClick={[Function]}
            style={
              Object {
                "color": "#333",
                "marginLeft": 24,
                "textDecoration": "none",
              }
            }
          >
            TECHNOLOGY
          </a>
        </span>
        <span
          className="MuiTypography-root-40 MuiTypography-button-51 MuiTypography-colorPrimary-60"
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <a
            href="/category/science"
            onClick={[Function]}
            style={
              Object {
                "color": "#333",
                "marginLeft": 24,
                "textDecoration": "none",
              }
            }
          >
            SCIENCE
          </a>
        </span>
        <span
          className="MuiTypography-root-40 MuiTypography-button-51 MuiTypography-colorPrimary-60"
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <a
            href="/category/art"
            onClick={[Function]}
            style={
              Object {
                "color": "#333",
                "marginLeft": 24,
                "textDecoration": "none",
              }
            }
          >
            ART
          </a>
        </span>
        <span
          className="MuiTypography-root-40 MuiTypography-button-51 MuiTypography-colorPrimary-60"
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <a
            href="/category/design"
            onClick={[Function]}
            style={
              Object {
                "color": "#333",
                "marginLeft": 24,
                "textDecoration": "none",
              }
            }
          >
            DESIGN
          </a>
        </span>
        <span
          className="MuiTypography-root-40 MuiTypography-button-51 MuiTypography-colorPrimary-60"
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <a
            href="/category/culture"
            onClick={[Function]}
            style={
              Object {
                "color": "#333",
                "marginLeft": 24,
                "textDecoration": "none",
              }
            }
          >
            CULTURE
          </a>
        </span>
        <span
          className="MuiTypography-root-40 MuiTypography-button-51 MuiTypography-colorPrimary-60"
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <a
            href="/category/photography"
            onClick={[Function]}
            style={
              Object {
                "color": "#333",
                "marginLeft": 24,
                "textDecoration": "none",
              }
            }
          >
            PHOTOGRAPHY
          </a>
        </span>
        <span
          className="MuiTypography-root-40 MuiTypography-button-51 MuiTypography-colorPrimary-60"
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <a
            href="/category/leadership"
            onClick={[Function]}
            style={
              Object {
                "color": "#333",
                "marginLeft": 24,
                "textDecoration": "none",
              }
            }
          >
            LEADERSHIP
          </a>
        </span>
        <span
          className="MuiTypography-root-40 MuiTypography-button-51 MuiTypography-colorPrimary-60"
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <a
            href="/category/math"
            onClick={[Function]}
            style={
              Object {
                "color": "#333",
                "marginLeft": 24,
                "textDecoration": "none",
              }
            }
          >
            MATH
          </a>
        </span>
        <span
          className="MuiTypography-root-40 MuiTypography-button-51 MuiTypography-colorPrimary-60"
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <a
            href="/category/economy"
            onClick={[Function]}
            style={
              Object {
                "color": "#333",
                "marginLeft": 24,
                "textDecoration": "none",
              }
            }
          >
            ECONOMY
          </a>
        </span>
        <span
          className="MuiTypography-root-40 MuiTypography-button-51 MuiTypography-colorPrimary-60"
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <a
            href="/category/music"
            onClick={[Function]}
            style={
              Object {
                "color": "#333",
                "marginLeft": 24,
                "textDecoration": "none",
              }
            }
          >
            MUSIC
          </a>
        </span>
        <span
          className="MuiTypography-root-40 MuiTypography-button-51 MuiTypography-colorPrimary-60"
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <a
            href="/categories"
            onClick={[Function]}
            style={
              Object {
                "color": "#333",
                "marginLeft": 24,
                "textDecoration": "none",
              }
            }
          >
            MORE
          </a>
        </span>
      </div>
    </header>
  </div>
  <div
    className="MuiGrid-container-65"
    style={
      Object {
        "flexDirection": "column",
        "marginLeft": "2%",
        "marginTop": "2%",
      }
    }
  >
    <h1
      className="MuiTypography-root-40 MuiTypography-headline-45 MuiTypography-gutterBottom-57"
    >
      technology
    </h1>
    <h2
      className="MuiTypography-root-40 MuiTypography-subheading-47 MuiTypography-gutterBottom-57"
    >
      Behind the scenes
    </h2>
  </div>
  <div
    className="MuiGrid-container-65 MuiGrid-spacing-xs-24-89"
  >
    <div
      className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-md-8-129"
      style={
        Object {
          "marginRight": 2,
        }
      }
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "marginTop": 24,
          }
        }
      >
        <div
          className="MuiPaper-root-9 MuiPaper-elevation1-12 MuiPaper-rounded-10 MuiCard-root-162"
          style={
            Object {
              "boxShadow": "0px 0.5px 0px 0px rgba(0, 0, 0, 0.2)",
              "display": "flex",
            }
          }
        >
          <div
            style={
              Object {
                "display": "flex",
                "flex": 4,
                "flexDirection": "column",
              }
            }
          >
            <div
              className="MuiCardContent-root-163"
              style={
                Object {
                  "flex": "1 0 auto",
                }
              }
            >
              <h2
                className="MuiTypography-root-40 MuiTypography-title-46"
                style={
                  Object {
                    "fontWeight": "bold",
                    "marginBottom": 8,
                  }
                }
              >
                abc
              </h2>
              <p
                className="MuiTypography-root-40 MuiTypography-body1-49"
              >
                test
              </p>
              <p
                className="MuiTypography-root-40 MuiTypography-body1-49"
                style={
                  Object {
                    "marginTop": 28,
                  }
                }
              >
                
              </p>
              <p
                className="MuiTypography-root-40 MuiTypography-body1-49 MuiTypography-colorTextSecondary-63"
                style={
                  Object {
                    "marginTop": 2,
                  }
                }
              >
                13 September 2018
              </p>
            </div>
          </div>
          <div
            style={
              Object {
                "display": "flex",
                "flex": 1,
                "justifyContent": "flex-end",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
